test/core.js

```diff
@@ -3,625 +3,601 @@
  */
 
 import assert from 'node:assert/strict'
-import {promises as fs, renameSync} from 'node:fs'
+import {promises as fs, renameSync, existsSync} from 'node:fs'
 import process from 'node:process'
 import {URL, pathToFileURL} from 'node:url'
 import test from 'node:test'
 import semver from 'semver'
 import {resolve} from '../index.js'
 
 const windows = process.platform === 'win32'
-const veryOldNode = semver.lt(process.versions.node, '16.0.0')
 const oldNode = semver.lt(process.versions.node, '18.0.0')
 
+const run = (/** @type {() => void} */ f) => f()
+
 process.on('exit', async () => {
   // Has to be sync.
-  renameSync('package.json.bak', 'package.json')
+  if (existsSync('package.json.bak')) {
```

Comment:
I’d prefer not to check this, but instead just try/catch it. AFAIK the docs still say that `exists*` and then doing something is an anti-pattern?

Reply:
Yeah, I know. It's in the docs, and also the reason why there's no `existsAsync`. But I really don't get the argument, and in the end it's irrelevant in this case because there are no multiple processes dealing with this file.

I can do it using a `try/catch`, but I think this is clearer. Your call. I prefer the existing code, but just say the word, and it changes to `try/catch`! 😊

Reply:
Yes, I would prefer just reading.
Either a simple one:

```
try {
  renameSync('package.json.bak', 'package.json')
} catch {}
```

Or a more involved:

```
try {
  renameSync("package.json.bak", "package.json");
} catch (error) {
  const exception = /** @type {ErrnoException} */ (error)
  // this next line is pseudo code: not sure what the exact code is off the top of my head
  if (exception.code === 'ERR_NOT_FOUND') return
  throw exception
}
```

Reply:
Used the more involved one.😊